Refactor/#42/게시물 api 수정

sim16065 opened the pull request:
관련 이슈
-
#이슈번호 (#42 )

📝 제목
-
[Refactor] 게시물 권한 처리 및 이미지 업로드 검증 수정

📋 작업 내용
-
- [ ]   이미지 업로드 URL 검증 추가 (base64 인코딩 및 jpg 포맷 확인)
- [ ]   유저 세션 기반으로 게시물 생성, 수정, 삭제 권한 처리 구현
- [ ]   검증 후 이미지 URL을 DB에 저장
- [ ]   utils 폴더 생성하여 중복되는  sendRespose 함수 분리.
- [ ]   클라이언트 측 에러 메세지 수정.
- [ ]   게시물 수정 시, 이미지를 반드시 제공할 필요가 없도록 수정.


🚀 테스트 사항(선택)
-
 기능 테스트 완료
 API 테스트 (Postman으로 이미지 업로드 테스트)
 예외 처리 확인 (base64와 jpg 포맷만 허용)

## 💬 리뷰 요구사항 (선택)  
이미지 업로드 로직에 대해 코드 리뷰 부탁드립니다.
예외 처리 부분에서 누락된 부분이 없는지 확인 부탁드립니다.

sim16065 committed:
setting/#39/cors-setting

sim16065 committed:
fix(#42): 게시물 권한 처리 및 이미지 업로드 검증 수정

- 유저 세션 기반으로 게시물 생성, 수정, 삭제 권한 처리 구현.
- 이미지 업로드 URL 검증 추가, base64 인코딩과 jpg 포맷 확인.
- 검증 후 이미지 URL을 DB에 저장.

gwgw123 commented on controllers/post.ctrl.js:
```diff
@@ -13,20 +14,20 @@ const sendResponse = (res, statusCode, success, message, data = null) => {
+        const imageUrl = await uploadBase64ImageToS3(postImg);
+              
+        const post = new Post({ content, postImg: imageUrl }, userId);
+        const postId = await post.createPost(); // createPost 호출 (Poststorage와 연결)
```
createPost 호출은 const Post = require("../models/Post"); 여기서 하는 중인데
주석 오타가 난 것인가요 ?

sim16065 replied:
주석 오타 맞습니다. 🥲
수정 완료했습니다~~

gwgw123 commented on controllers/post.ctrl.js:
```diff
@@ -48,16 +49,24 @@ const getAllPosts = async (req, res) => {
+
+        const result = await post.updatePost(id);
         
         if (result) {
```
```suggestion
        if (!result) {
          return sendResponse(res, 403, false, "수정 권한이 없거나 게시물이 존재하지 않습니다.");
```
와 같이 early return 방식을 사용하면 불 필요한 조건문 하나를 없앨 수 있어서
코드 가독성이 더 올라갈 것 같습니다~

sim16065 replied:
오 이해했습니다!

gwgw123 commented on controllers/post.ctrl.js:
```diff
@@ -48,16 +49,24 @@ const getAllPosts = async (req, res) => {
+            imageUrl = await uploadBase64ImageToS3(postImg);
+        }
+
+        const post = new Post({ content, postImg: imageUrl }, userId);
```
@Ryu-02 님한테도 말씀 드렸지만
userId를 DB에 사용하기전 사용자를 검증하는 로직이 필요해보입니다 ~

나중에 확인했는데 명규님이 만들고 계시군요 ㅎㅎ..

sim16065 committed:
refactor(#42): 코드 구조 개선 및 게시물 로직 수정

- `utils` 폴더 추가, `responseHelper` 분리
- Early return 적용
- 게시물 수정 시 기존 이미지 유지, 삭제 시 댓글도 삭제
- 클라이언트 에러 메시지 가독성 개선

sim16065 committed:
헬스체크

sim16065 committed:
헬스체크

sim16065 committed:
노드 버전 변경

sim16065 committed:
Merge branch 'develop' into fix/#42/post

sim16065 committed:
도커 파일 및 도메인 수정

sim16065 committed:
Merge branch 'fix/#42/post' of https://github.com/modern-agile-team/9term-mini-back into fix/#42/post

sim16065 committed:
좋아요 컨트롤러 추가
